### DISCUSSION TITLE
RSS 2.0 を利用した Feed の配信をしてほしい

### DISCUSSION THREAD

**Discussion by Code-Hex** [increments/qiita-discussions#95] [要望・アイデア・不具合]
簡単な比較が可能なように zenn.dev と qiita それぞれで配信しているフィードのフォーマットの例を用意しました。

- zenn.dev
  - https://zenn.dev/hajimehoshi/feed
- qiita
  - https://qiita.com/dodonki1223/feed

2.0 を利用できると例えば zenn.dev のように [enclosure](https://en.wikipedia.org/wiki/RSS_enclosure) というタグに OGP イメージの URL を付けることも可能になります。さらにブラウザ（Chrome 96.0.4664.110）で閲覧した場合ハイライトが効くのでとても見やすくなります。

**Comment by kskwtnk** [comment]
@Code-Hex 
ご意見ありがとうございます！

RSS 2.0に上げること自体は、多少時間はかかりそうですが可能です！
ただ、今私たちが考えていることの説明＆追加で質問をさせてください 🙏 

## 私たちが考えていることなど

- 今はATOMを使っているのもあり、RSS 2.0にすると変更量は大きいはず
- Qiita用のリーダーなどが作られていた場合、それが動かなくなってしまうかもしれない
- そうでなくとも、情報がガラッと変わることになるので各種RSSリーダーに全部「新規記事」扱いされてしまう可能性もなくはなさそう

## 追加の質問

- 例えばブラウザでのハイライトはどのような使用シーンがありますか？
- それ以外でRSS 2.0にしたい箇所はありますか？

要は、非破壊的な代替手段があればそちらを採用したい、と思っております 🙏

**Reply by Code-Hex** [reply]
@xrxoxcxox ありがとうございます！

現行の RSS と最新の RSS 2.0 の形式をある一定期間双方を配信し、1.0 -> 2.0 へのマイグレーションのアナウンスをするなど手順を踏めば既存のリーダーが動かなくなるインシデントは多少抑えられるかと思います。

> 例えばブラウザでのハイライトはどのような使用シーンがありますか？

どのタグをパースしようか考慮する際に利用します。

ブラウザで確認できるといちいちエディタを開かずに確認できるのでありがたいです。（開発環境が整ってないタブレット経由などでできる）

> それ以外でRSS 2.0にしたい箇所はありますか？

個人的には特にありませんが、ある程度他のテック情報配信系サービスと使用されるタグが統一されているとキュレーションサービスを作るときに便利だなとは感じます。

**Reply by kskwtnk** [reply]
@Code-Hex 
詳細にありがとうございます！

移行期間など含め、判断する箇所が多そうなので1度しっかりした議論を実施します！
その上でまたお返事いたしますので申し訳ありませんがしばらくお待ちください 🙇

**Reply by kskwtnk** [reply]
@Code-Hex 

返答が遅くなってしまい申し訳ないです :bow:

議論を実施しまして、結論としてはRSS 2.0への移行はしないことになりました
理由は以下のようになっています

- 不可逆的な変更になるため、マイグレーションのアナウンスなどを実施しても気付かず、対応しきれない方も多そう
    - これまでの各種変更の傾向から言うと、おそらくそれなりの数の方が気付かない
- もし実施する場合はATOMとRSS 2.0の両方を提供することになりそう
    - しかし現在の工数の状況を考えると実現が難しい

また、補足ですがQiita社としてATOMを選んだ理由は「拡張性が高い」ことです
任意の要素を追加したり、順序の表現ができたり……
現状あまりユニークなコンテンツを配信できていませんが、今後アドベントカレンダーなどの内容をATOMで柔軟にお届けしたいと考えております
更に言えば、今回ご意見をいただけた中にあるOGP画像の配信などもどこかで実施していければと思います

ご理解いただけますと幸いです！

**Reply by Code-Hex** [reply]
@xrxoxcxox ありがとうございます！
了解しました。楽しみにしています！
